app/scripts/modules/core/delivery/service/execution.service.js

```diff
@@ -38,6 +38,18 @@ module.exports = angular.module('spinnaker.core.delivery.executions.service', [
       return deferred.promise;
     }
 
+    function getExecution(executionId) {
+      var deferred = $q.defer();
+      const url = [ settings.gateUrl, 'pipelines', executionId].join('/');
+      $http({
+        method: 'GET',
+        url: url,
+        timeout: settings.pollSchedule * 2 + 5000, // TODO: replace with apiHost call
+      }).then((resp) => deferred.resolve(resp.data), deferred.reject);
+
+      return deferred.promise;
+    }
+
```

Comment:
Why manually build the promise from $q?   Just return the call to $http.  You can till extract the resp.data in the success block and don't have to worry about explicitly rejecting the deferred:  

    function getExecution(executionId) {
      const url = [ settings.gateUrl, 'pipelines', executionId].join('/');
      return $http({
        method: 'GET',
        url: url,
        timeout: settings.pollSchedule * 2 + 5000, // TODO: replace with apiHost call
      }).then((resp) => resp.data);
    }